Lib/test/test_typing.py

```diff
@@ -2864,7 +2864,7 @@ def test_get_type_hints_classes(self):
                          {'x': int, 'y': int})
         self.assertEqual(gth(mod_generics_cache.B),
                          {'my_inner_a1': mod_generics_cache.B.A,
-                          'my_inner_a2': mod_generics_cache.A,
+                          'my_inner_a2': mod_generics_cache.B.A,
```

Comment:
As far as possible, I tried to maintain backwards compatibility. To my surprise, this test failed, looking at it closer,  I think it might be wrong. Here's a snippet from the mod_generics_cache code
```python
 T = TypeVar('T')


    class A(Generic[T]):
        __annotations__ = dict(
            some_b='B'
        )


    class B(Generic[T]):
        class A(Generic[T]):
            pass

        __annotations__ = dict(
            my_inner_a1='B.A',
            my_inner_a2=A,
            my_outer_a='A'  # unless somebody calls get_type_hints with localns=B.__dict__
        )

```
``my_inner_a2`` looks to be referrring to ``B.A``, but because get_type_hints() previously only looked at globals, the test assumes ``A`` is correct, which seems wrong.